### PR TITLE
Send notification for specific MAS-related disabilities

### PR DESCRIPTION
**Pull request by allthesignals**
## Description of change
Adds a new Slack notification sent to the #rrd-to-mas-tracking tracking. Triggered when a MAS-specific contention is completed:

- prostate cancer (diagnostic code: 7528)
- sleep apnea (diagnostic code: 6847)
- rhinitis (diagnostic code: 6522)
- sinusitus (diagnostic codes: 6510, 6511, 6512, 6513, 6514)

Intended as the first step towards being able to pass claims to the Mail Automation System for processing.

## Original issue(s)
https://www.notion.so/Detect-single-issue-claims-for-increase-for-prostate-cancer-7528-sleep-apnea-6847-rhinitis-65-d9fda118dea5420a8c7fe9dabbbf1656

Closes #9909

## Things to know about this PR
There is an additional step that involves adding the slack email address to add the Slack notification email to the environment settings (rrd.mas_tracking.recipients).

This notification is wrapped with a Flipper feature in case it floods the channel.

## TODO
Identify best test strategy

### CODE REVIEW

File: app/services/rapid_ready_for_decision/constants.rb [hunk 1]
```diff
@@ -48,6 +48,9 @@ class Constants
 
     DISABILITIES_BY_CODE = DISABILITIES.map { |k, v| [v[:code], k] }.to_h
 
+    # prostate cancer (1), sleep apnea (1), rhinitis (1), sinusitus (4)
+    MAS_DISABILITIES = [7528, 6847, 6522, 6510, 6511, 6512, 6513, 6514].freeze
+
```
Comment (yoomlam) [inline app/services/rapid_ready_for_decision/constants.rb:53]: > Should MAS go into the constants class?

This is fine. It would be better if you grouped the DCs maybe on separate lines:
```suggestion
    MAS_DISABILITIES = [
      7528, # prostate cancer
      6847, # sleep apnea
      6522, # rhinitis
      6510, 6511, 6512, 6513, 6514 #sinusitus
    ].freeze
```

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb [hunk 2]
```diff
@@ -72,6 +88,14 @@ def rrd_special_issue_set?
     end
   end
 
+  # TODO: extract into constants code
```
Comment (yoomlam) [inline app/models/concerns/form526_rapid_ready_for_decision_concern.rb:91]: Don't know what you mean by this TODO.

File: config/settings.yml [hunk 3]
```diff
@@ -1317,6 +1317,9 @@ rrd:
   event_tracking:
     recipients:
       - fake_email
+  mas_tracking:
+    recipients:
+      - fake_email
```
Comment (yoomlam) [inline config/settings.yml:1322]: Where does the real email setting go?

File: config/initializers/bgs.rb [hunk 4]
```diff
@@ -6,7 +6,7 @@
   # OpenSSL::SSL::VERIFY_PEER = OpenSSL::SSL::VERIFY_NONE
   BGS.configure do |config|
     config.application = Settings.bgs.application
-    config.client_ip = Socket.ip_address_list.detect(&:ipv4_private?).ip_address
+    config.client_ip = Socket.ip_address_list.detect(&:ipv4_private?)
```
Comment (yoomlam) [inline config/initializers/bgs.rb:9]: Why is this changed?

**Review by yoomlam — COMMENTED** [review]
Comments on WIP.

> Should things go in the concern? Or should detection logic go in the job selector (submission.rrd_job_selector.rrd_applicable?)?
> Should MAS go into the constants class?
> Should MAS disability code logic be moved from the concern into the Constants as a class method?

You changes look fine.

> The message itself will need to include the diagnostic code. Should this re-use some class method logic in Constants to extract these codes?

I'd add `diagnostic_codes` to the `Form526RapidReadyForDecisionConcern` class.

File: config/settings.yml (hunk 3 above)
Comment (allthesignals) [reply config/settings.yml:1322]: @yoomlam prod?

File: config/initializers/bgs.rb (hunk 4 above)
Comment (allthesignals) [reply config/initializers/bgs.rb:9]: @yoomlam good catch, my network settings are funky. i didn't meant to commit. ty.

File: config/settings.yml (hunk 3 above)
Comment (allthesignals) [reply config/settings.yml:1322]: @yoomlam i assume it goes in the environment of prod. seems to follow the same pattern of other slack notifications

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb [hunk 5]
```diff
@@ -19,6 +19,22 @@ def send_rrd_alert_email(subject, message, error = nil)
...
+      Form526Submission.id: #{id}<br/>
+      <br/>
+      #{message}<br/>
+    BODY
+    ActionMailer::Base.mail(
+      from: ApplicationMailer.default[:from],
+      to: Settings.rrd.mas_tracking.recipients,
+      subject: "MA claim - #{id}", # TODO: get diagnostiCode
+      body: body
+    ).deliver_now
```
[the remark this answers is not in the document]
Comment (yoomlam) [reply app/models/concerns/form526_rapid_ready_for_decision_concern.rb:35]: Agree

File: app/workers/evss/disability_compensation_form/submit_form526.rb [hunk 6]
```diff
@@ -40,6 +40,7 @@ def perform(submission_id)
           response_handler(response)
         end
         send_rrd_completed_notification(submission) if submission.rrd_job_selector.rrd_applicable?
+        submission.send_rrd_to_mas_email if submission.rrd_to_mas_disability?
```
[the remark this answers is not in the document]
Comment (yoomlam) [reply app/workers/evss/disability_compensation_form/submit_form526.rb]: Ah, I see what you're saying. In that case, maybe rename `rrd_to_mas_disability?` to `forward_to_mas?` and `send_rrd_to_mas_email` to `send_to_mas_email`. The `rrd` throws me off. Since these claims are not `rrd_applicable?`, then let's remove `rrd` from the method names. Should also add in comments what `mas` means.

[the remark this answers is not in the document]
Comment (allthesignals) [reply app/workers/evss/disability_compensation_form/submit_form526.rb]: @yoomlam yep, good point, that's my bad: rrd in the name is confusing! Okay, will work on better naming. Tyvm.

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb [hunk 7]
```diff
@@ -72,6 +97,18 @@ def rrd_special_issue_set?
...
+    form_disabilities.map { |disability| disability['diagnosticCode'] }
+  end
+
+  # TODO: extract into constants code
+  def forward_to_mas?
+    RapidReadyForDecision::Constants::MAS_DISABILITIES.any? do |i|
+      diagnostic_codes.include?(i)
+    end
+  end
+
```
[the remark this answers is not in the document]
Comment (allthesignals) [reply app/models/concerns/form526_rapid_ready_for_decision_concern.rb:111]: makes sense, i could mimic how `.rrd_applicable?` gates stuff

**Comment by allthesignals** [comment]
> I think it might make sense to separate from the rrd_alert_email, which using that doesn't seem necessary, but don't consider that blocking.

Agree, will push changes up. ty

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb [hunk 8]
```diff
@@ -72,6 +96,18 @@ def rrd_special_issue_set?
...
+  def diagnostic_codes
+    form_disabilities = form.dig('form526', 'form526', 'disabilities')
+    form_disabilities.map { |disability| disability['diagnosticCode'] }
+  end
+
+  # TODO: extract into constants code
+  def forward_to_mas?
+    RapidReadyForDecision::Constants::MAS_DISABILITIES.any? do |i|
+      diagnostic_codes.include?(i)
+    end && diagnostic_codes.size == 1
```
Comment (yoomlam) [inline app/models/concerns/form526_rapid_ready_for_decision_concern.rb:108]: I would call `diagnostic_codes.size == 1` first since it's a faster operation.

This:
```ruby
    RapidReadyForDecision::Constants::MAS_DISABILITIES.any? do |i|
      diagnostic_codes.include?(i)
    end
```
could be shortened and easier to comprehend if you did set-intersection or subtracting arrays to determine if any element in one array is an element in another array. Even better, since you know there's only 1 element in `diagnostic_codes` (after checking for `diagnostic_codes.size == 1`), just call `MAS_DISABILITIES.include?(diagnostic_codes.first)`.

File: spec/jobs/evss/disability_compensation_form/submit_form526_all_claim_spec.rb [hunk 9]
```diff
@@ -45,6 +45,24 @@ def expect_retryable_error(error_class)
...
+
+        it 'sends an email for tracking purposes' do
+          VCR.use_cassette('evss/disability_compensation_form/submit_form_v2') do
+            subject.perform_async(submission.id)
+            described_class.drain
+            expect(ActionMailer::Base.deliveries.last.subject).to eq 'MA claim - 6847'
+          end
+        end
+      end
     end
```
Comment (yoomlam) [inline spec/jobs/evss/disability_compensation_form/submit_form526_all_claim_spec.rb:66]: Probably add a test with multiple MAS-related DCs and check that a notification is not sent.

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb [hunk 10]
```diff
@@ -19,6 +19,30 @@ def send_rrd_alert_email(subject, message, error = nil)
     ).deliver_now
   end
 
+  def send_to_mas_email
+    message = <<~BODY
+      <table border="1" cellspacing="1" cellpadding="5">
```
[the remark this answers is not in the document]
Comment (allthesignals) [reply app/models/concerns/form526_rapid_ready_for_decision_concern.rb:24]: Okay!

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb [hunk 11]
```diff
@@ -19,6 +19,31 @@ def send_rrd_alert_email(subject, message, error = nil)
...
+            <td>#{submitted_claim_id}</td>
+            <td>#{created_at.to_date}</td>
+            <td>#{created_at.strftime '%H:%M:%S'}</td>
+            <td>#{id}</td>
+          </tr>
+        </tbody>
+      </table>
+    BODY
+
+    send_rrd_alert_email("MA claim - #{diagnostic_codes.join(',')}", message, nil, Settings.rrd.mas_tracking.recipients)
```
[the remark this answers is not in the document]
Comment (allthesignals) [reply app/models/concerns/form526_rapid_ready_for_decision_concern.rb:43]: Agree

[the remark this answers is not in the document]
Comment (allthesignals) [reply app/models/concerns/form526_rapid_ready_for_decision_concern.rb:43]: Right, it only sends for single-issue now... probably not relevant anymore...

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb (hunk 8 above)
Comment (allthesignals) [reply app/models/concerns/form526_rapid_ready_for_decision_concern.rb:108]: Oh right, since it's only relevant if it's single-issue. thanks!

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb (hunk 7 above)
[the remark this answers is not in the document]
Comment (leikkisa) [reply app/models/concerns/form526_rapid_ready_for_decision_concern.rb:110]: Here's an example of where that logic is currently, they could be used as a basis for adding the logic to `forward_to_mas?`
`SidekiqJobSelector.disability_increase?` and `RrdSpecialIssueManager.included_disability_increase?`

[the remark this answers is not in the document]
Comment (allthesignals) [reply app/models/concerns/form526_rapid_ready_for_decision_concern.rb:110]: This was addressed in method `forward_to_mas?`

```ruby
  def forward_to_mas?
    # only use the first diagnostic code because we can only support single-issue claims
    diagnostic_codes.size == 1 && RapidReadyForDecision::Constants::MAS_DISABILITIES.include?(diagnostic_codes.first)
  end
```

File: spec/jobs/evss/disability_compensation_form/submit_form526_all_claim_spec.rb (hunk 9 above)
Comment (allthesignals) [reply spec/jobs/evss/disability_compensation_form/submit_form526_all_claim_spec.rb:66]: done

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb [hunk 12]
```diff
@@ -13,12 +13,37 @@ def send_rrd_alert_email(subject, message, error = nil)
...
-      to: Settings.rrd.alerts.recipients,
+      to: to,
       subject: subject,
       body: body
     ).deliver_now
   end
 
+  def send_to_mas_email
+    message = <<~BODY
+      #{disabilities.pluck('name').join(', ')}
```
Comment (yoomlam) [inline app/models/concerns/form526_rapid_ready_for_decision_concern.rb:24]: Suggest including the DC id for each name.

**Review by yoomlam — APPROVED** [review]
Lookin' good! Ticket description needs cleanup. Left a nonblocking suggestion.

File: config/settings.yml (hunk 3 above)
Comment (allthesignals) [reply config/settings.yml:1322]: "# Add a new feature toggle here to ensure that it is initialized in all environments." in config/features.yml.

I expect this to be available in the Flipper configuration pages when deployed

Comment (yoomlam) [reply config/settings.yml:1322]: I was referring to the email address that will send it to Slack.
